pkg/util/util.go

```diff
@@ -30,6 +30,70 @@ const (
 	osdServiceAccountName = "osd-managed-admin"
 )
 
+// ConfigMapOperations store informations required for operations
+type ConfigMapOperations struct {
+	name       string
+	namespace  string
+	kubeClient client.Client
+}
+
+// GetConfigMapOperations returns a new ConfigMapOperations object
+func GetConfigMapOperations(kubeClient client.Client, name, namespace string) *ConfigMapOperations {
+	config := ConfigMapOperations{
+		name:       name,
+		namespace:  namespace,
+		kubeClient: kubeClient,
+	}
+
+	return &config
+}
+
+// getConfigMap returns a configmap
+func (c *ConfigMapOperations) getConfigMap() (*corev1.ConfigMap, error) {
+	cfg := &corev1.ConfigMap{}
+	if err := c.kubeClient.Get(context.TODO(), kubetypes.NamespacedName{Name: c.name, Namespace: c.namespace}, cfg); err != nil {
+		return &corev1.ConfigMap{}, err
+	}
+
+	return cfg, nil
+}
+
+// getValue returns value if the key exists in configmap
+func (c *ConfigMapOperations) getValue(key string) (string, error) {
+	configmap, err := c.getConfigMap()
+	if err != nil {
+		return "", fmt.Errorf("clusterdeployment.GetGCPParentFolderFromConfigMap.Get %v", err)
+	}
+
+	value, ok := configmap.Data[key]
+	if !ok {
+		return "", fmt.Errorf("GCP configmap %v did not contain key %v",
+			c.name, key)
+	}
+
+	return value, nil
+}
+
+// GetParentFolder returns orgParentFolderID if the key exists in configmap
+func (c *ConfigMapOperations) GetParentFolder() (string, error) {
+	value, err := c.getValue("orgParentFolderID")
+	if err != nil {
```

Comment:
I don't think you need to wrap that error here again... you could just return `value, err`

Reply:
done